### PR TITLE
Application Architecture

### LINKED ISSUE #233: Application Architecture

### PR DESCRIPTION
**Pull request by hieuk09**
[Close #233]

## Why
- extraction of a microservice
- for when user need a small web app where they would normally use Sinatra.

### CLI
```shell
% lotus new bookshelf --architecture=app
```

### Output

Basically like the combine of these commands:
```shell
% lotus new bookshelf && cd bookshelf && bundle
% lotus generate app bookshelf
% mv apps/bookshelf ..
```

### Bonus

In an application with application architecture, you don't need to specify container name when using action generator:

```shell
# using
lotus generate action dashboard#index
# instead of 
lotus generate action web dashboard#index
```

### CODE REVIEW

File: test/commands/new_test.rb [hunk 1]
```diff
@@ -825,9 +826,844 @@ def container_options
...
+    def clear_root_folder
+      FileUtils.rm_rf("#{@root}/.", secure: true)
+    end
+
+    let(:opts)      { container_options }
+    let(:app_name)  { '.' }
+
+    before do
+      clear_root_folder
+      capture_io { command.start }
```
Comment (hieuk09) [inline test/commands/new_test.rb]: I still have problem with the tests here. I'm not sure why `.lotusrc` is created before the generator is called here. It makes the tests stucked waiting for user input. Could you give me some insight about this? 
Cc: @jodosha, @joneslee85

Comment (joneslee85) [inline test/commands/new_test.rb]: @hieuk09 I could pair with you 2moro night (at hacknight over ScreenHero)

Comment (hieuk09) [inline test/commands/new_test.rb]: I figured out the bug :dancer:. Thanks anyway :)

**Comment by jodosha** [comment]
@hieuk09 Thanks for starting this. There was a misunderstanding between us. Here what I wanted to achieve:

```shell
lotus new ping --architecture=app && cd ping
tree
.
├── Gemfile
├── Rakefile
├── app
│   ├── controllers
│   ├── templates
│   │   └── application.html.erb
│   └── views
│       └── application_layout.rb
├── config
│   ├── application.rb
│   ├── environment.rb
│   └── routes.rb
├── config.ru
├── db
├── lib
│   ├── config
│   │   └── mapping.rb
│   ├── ping
│   │   ├── entities
│   │   └── repositories
│   └── ping.rb
├── public
│   ├── javascripts
│   └── stylesheets
└── spec
    ├── controllers
    ├── features
    ├── features_helper.rb
    ├── ping
    │   ├── entities
    │   └── repositories
    ├── spec_helper.rb
    ├── support
    └── views

22 directories, 12 files
```

Here's the output for action generator:

```shell
% lotus generate action home#index
      insert  config/routes.rb
      create  spec/controllers/home/index_spec.rb
      create  app/controllers/home/index.rb
      create  app/views/home/index.rb
      create  app/templates/home/index.html.erb
      create  spec/views/home/index_spec.rb
```

I'm testing it manually because there are still a lot of issues.

**Comment by hieuk09** [comment]
Sorry about the misunderstanding, @jodosha. I've just tried to amend the PR and I saw that you already finished fixing it on master :)

**Pull request merged**